### ISSUE TITLE
TypeScriptLexer cannot lex template strings

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#997]
(Original issue 1293 created by None on 2016-10-20T21:16:16.296947+00:00)

Here's a simple reproduction, using `raiseonerror` to make it fairly obvious:

```
#!python

>>> lexer = TypeScriptLexer()
>>> lexer.add_filter("raiseonerror")
>>>
>>> doc = """
... let x = `this is valid typescript`;
... """
>>>
>>> list(lex(doc, lexer))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/dcrosta/.virtualenvs/platform-guild/lib/python2.7/site-packages/pygments/filter.py", line 20, in _apply
    for token in filter_.filter(lexer, stream):
  File "/Users/dcrosta/.virtualenvs/platform-guild/lib/python2.7/site-packages/pygments/filters/__init__.py", line 196, in filter
    raise self.exception(value)
pygments.filters.ErrorToken: `
```

I'll be happy to submit a patch for this if someone can suggest where I might start digging -- I'm not very familiar with Pygments' internals.

**Comment by Anteru** [comment]
(Original comment by None on 2016-10-20T21:17:26.247595+00:00)

Err, that was me who submitted this, didn't realize I wasn't logged in :-(

**Comment by Anteru** [comment]
(Original comment by None on 2016-10-21T18:21:41.847873+00:00)

Fixed in pull request #677
